protoc-gen-pbandk/lib/src/commonMain/kotlin/pbandk/gen/Namer.kt

```diff
@@ -52,11 +52,14 @@ public interface Namer {
 
         override fun newEnumValueTypeName(enumTypeName: String, preferred: String, nameSet: Collection<String>): String {
             val typePrefix = splitWordsToSnakeCase(enumTypeName) + '_'
-            var name = splitWordsToSnakeCase(preferred).substringAfter(typePrefix)
-            name = name.uppercase()
+            var name = splitWordsToSnakeCase(preferred).let { snakeCase ->
+                snakeCase.substringAfter(typePrefix).takeUnless {
+                    "^\\d|\\W".toRegex().matches(it)
+                } ?: snakeCase
+            }.apply { uppercase() }
```

Comment:
You'll need to use `let` instead of `apply` I think. `uppercase()` returns a copy of the string rather than modifying it in-place. So `apply { uppercase() }` has no effect since `apply` returns the original string rather than the copy returned by `uppercase()`.